### PR TITLE
Add FinancialManagement

### PR DESCRIPTION
**Pull request by linhvnguyen9**

### CODE REVIEW

File: contracts/FinancialManagement.sol [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+contract UserManagement {
+    address public admin;
+    address public manager;
+
+    constructor() {
+        admin = msg.sender;
+    }
+
+    modifier onlyAdmin {
+        require(msg.sender == admin);
```
Comment (conalpham) [inline contracts/FinancialManagement.sol]: We should return a message error :v

File: contracts/FinancialManagement.sol [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+contract FinancialOperations is UserManagement {
+    mapping (address => uint) public deposits;
+
+    function deposit() payable onlyUser public {
+        deposits[msg.sender] += msg.value;
+    }
+
+    function withdraw(uint amount) public {
+        require(deposits[msg.sender] >= amount, "Withdrawing greater than deposited amount!");
+        require(address(this).balance >= amount, "Bank run :(");
```
Comment (conalpham) [inline contracts/FinancialManagement.sol:45]: Great!!!! =)))))

File: contracts/FinancialManagement.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        require(address(this).balance >= loanRequests[msg.sender].amount, "No money in bank for you to get loan :(");
+        require(loanRequests[msg.sender].withdrawn == false, "You've already got your loan!");
+        loanRequests[msg.sender].withdrawn = true;
+        (bool sent, bytes memory data) = msg.sender.call{ value: loanRequests[msg.sender].amount }("");
+        require(sent, "Failed to send Ether");
+    }
+
+    function repay() payable public onlyUser {
+        if (msg.value > loanRequests[msg.sender].amount) {
+            loanRequests[msg.sender].amount = 0; // Thanks for the donation!
```
Comment (conalpham) [inline contracts/FinancialManagement.sol:85]: i think we should refund dust amount to user :v

File: contracts/FinancialManagement.sol [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    function deposit() payable onlyUser public {
+        deposits[msg.sender] += msg.value;
+    }
+
+    function withdraw(uint amount) public {
+        require(deposits[msg.sender] >= amount, "Withdrawing greater than deposited amount!");
+        require(address(this).balance >= amount, "Bank run :(");
+        (bool sent, bytes memory data) = msg.sender.call{ value: amount }("");
+        require(sent, "Failed to send Ether");
+        deposits[msg.sender] -= amount;
```
Comment (conalpham) [inline contracts/FinancialManagement.sol]: This code can be attacked by Reentrancy Attack :v  You should push this line before line 46 :v

File: contracts/FinancialManagement.sol (hunk 1 above)
Comment (linhvnguyen9) [reply contracts/FinancialManagement.sol]: thanks for the suggestion! Updated

File: contracts/FinancialManagement.sol (hunk 3 above)
Comment (linhvnguyen9) [reply contracts/FinancialManagement.sol:85]: updated now

File: contracts/FinancialManagement.sol (hunk 4 above)
Comment (linhvnguyen9) [reply contracts/FinancialManagement.sol]: yep, updated now

**Pull request merged**